File: May-LeetCoding-Challenge/01-First-Bad-Version/First-Bad-Version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int left = 1, right = n;
        while (left < right){
            int mid = (right - left) / 2 + left;
            if (isBadVersion(mid))
                right = mid;
            else left = mid + 1;
        }
        return left;
    }
};

File: May-LeetCoding-Challenge/01-First-Bad-Version/First-Bad-Version.cs
/* The isBadVersion API is defined in the parent class VersionControl.
      bool IsBadVersion(int version); */

public class Solution : VersionControl {

  public int FirstBadVersion(int n) {
		int left = 1;
		int right = n;
		while (left < right) {
			int mid = left + (right - left) /2;
			if (IsBadVersion(mid)){
				right = mid;
			}
			else 
			{
				left = mid + 1;
			}
		}
		return left;        
	}
   }
